src/main/java/com/medallia/word2vec/neuralnetwork/NeuralNetworkTrainer.java

```diff
@@ -151,28 +156,30 @@ static long incrementRandom(long r) {
 	}
 	
 	/** @return Trained NN model */
-	public NeuralNetworkModel train(Iterable<List<String>> sentences) throws InterruptedException {
-		ListeningExecutorService ex = MoreExecutors.listeningDecorator(Executors.newFixedThreadPool(config.numThreads));
-		
+	public NeuralNetworkModel train(final Iterable<List<String>> sentences) throws InterruptedException {
+		final ListeningExecutorService ex =
+				MoreExecutors.listeningDecorator(
+					new ThreadPoolExecutor(config.numThreads, config.numThreads,
+							0L, TimeUnit.MILLISECONDS,
+							new ArrayBlockingQueue<Runnable>(8),
+							new ThreadPoolExecutor.CallerRunsPolicy()));
+
 		int numSentences = Iterables.size(sentences);
 		numTrainedTokens += numSentences;
-		
-		// Partition the sentences evenly amongst the threads
-		Iterable<List<List<String>>> partitioned = Iterables.partition(sentences, numSentences / config.numThreads + 1);
-		
+
+		// Partition the sentences into batches
+		final Iterable<List<List<String>>> batched = Iterables.partition(sentences, 1024);
```

Comment:
Hmm, I played around with this during implementation and IIRC, it actually does have an effect on the output word vectors since the learning rate is updated concurrently by the workers ...

Can you explain why this is necessary?  Maybe we can add a flag to toggle behavior on it if it's truly necessary.